### ISSUE TITLE
Low contrast between background colour and excluded folders

### ISSUE THREAD

**Issue by sushisharkjl** [arcticicestudio/nord-jetbrains#57]
Deselected excluded folders have really low contrast that makes their names difficult to read:

![image](https://user-images.githubusercontent.com/13944989/58767851-ee5fbc80-8545-11e9-8a7c-0704d355ae82.png)

Once highlighted, their appearance improves:

![image](https://user-images.githubusercontent.com/13944989/58767863-28c95980-8546-11e9-8c59-602158d05ee8.png)

**Comment by arcticicestudio** [comment]
Hi @sushisharkjl :wave:, thanks for your contribution :+1:
I guess this is a problem based on two factors: The foreground color for excluded files and folder uses a low contrast by design. This decision has been made so the theme assists you to keep the focus on the important things: the source code as well as the relevant files/folders when seen for the tree view. The excluded data is excluded with a reason: you shouldn't care about it :smile: 

Anyway, the second fact is that it looks like you're using IntelliJ's [_File Colors_][fc] features that highlights some files in various UI elements with different background colors. You can take a look at #53 that changed the style for this features (release in version [0.6.0][]) and also follow the instructions in the linked JetBrains documentation to disable the features if you like. I've never used it and in my opinion the UI feel way more cleaner + the files/folders can be read better again.

Let me know if it helps to make it useable again for you :+1: 

[fc]: https://www.jetbrains.com/help/idea/settings-file-colors.html
[0.6.0]: https://github.com/arcticicestudio/nord-jetbrains/releases/tag/v0.6.0

**Comment by Sushisource** [comment]
I'll chip in here and say this struck me as a bit painful too. I understand the philosophy, but given that File Colors is enabled by default, probably makes sense to make these a little bit more legible. De-emphasizing makes perfect sense, but not to the point of nearly-impossible-to-read. Personally, I find the highlights quite useful for test folders.

The fix in #53 doesn't seem to be totally working, as I am using 0.6.0 and get the exact look as in OP's screenshot. I would expect the text on the excluded folders to be white as in the fix, but it's still greyish for me.

Additionally, the little expansion arrows are really really hard to see on top of that blue background when items are highlighted.

Love your theme! Thanks for your hard work!

**Comment by arcticicestudio** [comment]
@Sushisource I totally agree, but unfortunately there are no theme keys to explicitly set the foreground color of ignored/excluded files in a tree view when a file color has been applied. The only solution would be to change the general ignore/exclude foreground color which again would make it impossible to differ between a normal and excluded folder when the user has disabled the _File Colors_ feature. Also even when _File Colors_ are enabled by default, the **background color is not inherited from the active UI theme** but the default and „hardcoded“ values of the parent theme _Darcula_.

In short: As long as there is no way to apply specific colors/styles when a specific condition is met this is a edge case that a theme author can not handle.

There are other edge case that are hard to handle since the **user can choose any background color**.
How to determine if the chosen background color is dark or bright? Using a bright color like white in combination with e.g. `nord4` as foreground color would make text almost invisible. On the other hand in situations like now where the default/fallback _Darcula_ values are used the dark foreground color `nord3` makes it also hard to read.

This pseudo code shows an possible solution that is necessary in order to fix the problem that must be added by JetBrains to the theme API in any way exposed to the JSON theme interface:

```go
var ignore_or_excluded_foreground_color = "nord3"
var matching_file_color_scope_background_color = getMatchingFileColorBgColor()

if file_colors {
  if calcBrightnessRatio(matching_file_color_scope_background_color) > "50%" {
    ignore_or_excluded_foreground_color = "nord3"
  } else {
     ignore_or_excluded_foreground_color = "nord3"
  }
}
```

I hope this makes the situation a bit more clear, it's not as easy as you think and unfortunately as a theme author I depend on a solution on the side of JetBrains :disappointed:

**Comment by jgrossi** [comment]
@sushisharkjl @Sushisource I have the same issue. the only solution was unchecking the "ignored" "file status color":
<img width="1278" alt="image" src="https://user-images.githubusercontent.com/1175275/63944402-d49c1b80-ca47-11e9-9a2b-ddcbb35f351c.png">

**Comment by arcticicestudio** [comment]
@jgrossi Oh, I never saw this screen in previous IDE versions before :astonished: 
I'll definitely need to check this. The theme scheme itself doesn't expose any keys to customize the colors so I'll initialize and commit all IDE setting into a Git repository and test which keys are added when using this UI view :smile:

**Comment by adampatterson** [comment]
Disabling "File Colors" solved it for me, I was able to leave the Version status colors as is.

![image](https://user-images.githubusercontent.com/31537/68088428-6a736e80-fe1c-11e9-9f9b-7cbf8c72a744.png)

**Comment by r4j4h** [comment]
not sure where this should go but it seems related to this, so I wanted to emphasize this proposed solution on the official plugins reviews: https://imgur.com/r7G5WgX from https://plugins.jetbrains.com/plugin/10321-nord/reviews#review=53562-53563

**Comment by xsor-hbo** [comment]
A floating, ghost `library root` looks very scary with the no-contract text + background color 😐 
![Image](https://github.com/user-attachments/assets/82122f30-d9c3-4be8-b05b-bef53a12ec17)

**Comment by ruchernchong** [comment]
> A floating, ghost `library root` looks very scary with the no-contract text + background color 😐 ![Image](https://github.com/user-attachments/assets/82122f30-d9c3-4be8-b05b-bef53a12ec17)

I have the exact same visuals as you
